src/components/minter/mint.cairo

```diff
@@ -0,0 +1,287 @@
+        cc_distributed: Span<u256>,
+    }
+
+    mod Errors {
+        const INVALID_ARRAY_LENGTH: felt252 = 'ERC1155: invalid array length';
+    }
+
+    #[embeddable_as(MintImpl)]
+    impl Mint<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>
+    > of IMint<ComponentState<TContractState>> {
+        fn get_carbonable_project_address(
+            self: @ComponentState<TContractState>
+        ) -> ContractAddress {
+            self.Mint_carbonable_project_address.read()
+        }
+
+        fn get_payment_token_address(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Mint_payment_token_address.read()
+        }
+
+        fn is_public_sale_open(self: @ComponentState<TContractState>) -> bool {
+            self.Mint_public_sale_open.read()
+        }
+
+        fn get_unit_price(self: @ComponentState<TContractState>) -> u256 {
+            self.Mint_unit_price.read()
+        }
+
+        fn get_available_money_amount(self: @ComponentState<TContractState>) -> u256 {
+            self.Mint_remaining_money_amount.read()
+        }
+
+        fn get_min_money_amount_per_tx(self: @ComponentState<TContractState>) -> u256 {
+            self.Mint_min_money_amount_per_tx.read()
+        }
+
+        fn is_sold_out(self: @ComponentState<TContractState>) -> bool {
+            self.get_available_money_amount() == 0
+        }
+
+        fn is_canceled(self: @ComponentState<TContractState>) -> bool {
+            self.Mint_cancel.read()
+        }
+
+        fn set_public_sale_open(ref self: ComponentState<TContractState>, public_sale_open: bool) {
+            // [Effect] Update storage
+            self.Mint_public_sale_open.write(public_sale_open);
+
+            // [Event] Emit event
+            let current_time = get_block_timestamp();
+            if public_sale_open {
+                self.emit(PublicSaleOpen { time: current_time });
+            } else {
+                self.emit(PublicSaleClose { time: current_time });
+            };
+        }
+
+        fn set_unit_price(ref self: ComponentState<TContractState>, unit_price: u256) {
+            // [Check] Value not null
+            assert(unit_price > 0, 'Invalid unit price');
+            // [Effect] Store value
+            self.Mint_unit_price.write(unit_price);
+        }
+
+        fn withdraw(ref self: ComponentState<TContractState>) {
+            // [Compute] Balance to withdraw
+            let token_address = self.Mint_payment_token_address.read();
+            let erc20 = IERC20CamelDispatcher { contract_address: token_address };
+            let contract_address = get_contract_address();
+            let balance = erc20.balanceOf(contract_address);
+
+            // [Interaction] Transfer tokens
+            let caller_address = get_caller_address();
+            let success = erc20.transfer(caller_address, balance);
+            assert(success, 'Transfer failed');
+        }
+
+        fn transfer(
```

Comment:
Generally not a good idea to execute contracts from outside.
I assume this method be protected by Access Control later, but right now it's not.
Also, `transfer` might not be a precise enough name and clashes with the name of ERC20 methods.

Reply:
Agreed and need to add Ownership control when we will add Access Control.